testrunner-running/src/main/scala/edu/illinois/cs/testrunner/data/framework/TestFramework.scala

```diff
@@ -105,19 +106,44 @@ object JUnit extends TestFramework {
 
 object JUnit5 extends TestFramework {
     val methodAnnotationStr: String = "org.junit.jupiter.api.Test"
+    val nestedAnnotationStr: String = "org.junit.jupiter.api.Nested"
+    val disabledAnnotationStr: String = "org.junit.jupiter.api.Disabled"
 
     override def tryGenerateTestClass(loader: ClassLoader, clzName: String)
             : Option[GeneralTestClass] = {
-        val annotation: Class[_ <: Annotation] =
+        val testAnnotation: Class[_ <: Annotation] =
             loader.loadClass(methodAnnotationStr).asInstanceOf[Class[_ <: Annotation]]
 
+        val disabledAnnotation: Class[_ <: Annotation] =
+            loader.loadClass(disabledAnnotationStr).asInstanceOf[Class[_ <: Annotation]]
+
         try {
             val clz = loader.loadClass(clzName)
 
+            if (clz.getAnnotation(disabledAnnotation) != null) {
+                // skip disabled class
+                return Option.empty
+            }
+
+            if (clz.isMemberClass) {
+                val nestedAnnotation: Class[_ <: Annotation] =
+                        loader.loadClass(nestedAnnotationStr)
+                              .asInstanceOf[Class[_ <: Annotation]]
+                if (Modifier.isStatic(clz.getModifiers) ||
```

Comment:
Is it such that JUnit 5 should not treat a static nested class that is not annotated with @Nested as a valid test class? The example presented in Issue #12 seems to be a nested static class that is not annotated.

Reply:
I check the official doc again, the nested test class should be annotated with `@Nested` and be `non-static`. Any violation will cause `mvn test` fail to run the tests.

I try to run the example in #12. Actually `mvn test` doesn't run that test. That's a bad example and there is some other problem with that test.

Reply:
Thanks for double-checking. Would these changes resolve that Issue #12, where it's just a wrong example there, or is there some other problem present in Issue #12 that's not handled yet?

Reply:
These changes resolve Issue #12 

BTW, all the solutions will be verified with my demo project first, like the [nested class test](https://github.com/luoos/junit5-testrunner-demo/blob/f3012d1b8652cca1780f9d43c40e4524d975bf24/src/test/java/com/luojl/demo/TestRunnerTest.java#L246)